### ISSUE TITLE
False positives in destructuring-assignment

### ISSUE THREAD

**Issue by ThiefMaster** [yannickcr/eslint-plugin-react#3119]
Not sure if it's caused by the changes in #3001, but after updating from 7.24 to 7.26, I noticed a bunch of false positives showing up in my code:

```
.../indico/modules/events/editing/client/js/editing/timeline/util.js
  55:14  warning  Must use destructuring revision assignment  react/destructuring-assignment

.../indico/modules/rb/client/js/common/map/reducers.js
  59:9  warning  Must use destructuring action assignment  react/destructuring-assignment
  64:9  warning  Must use destructuring action assignment  react/destructuring-assignment
```

- https://github.com/indico/indico/blob/676dc3d1fede9bd4417982e79310753691be2472/indico/modules/rb/client/js/common/map/reducers.js#L59
- https://github.com/indico/indico/blob/676dc3d1fede9bd4417982e79310753691be2472/indico/modules/rb/client/js/common/map/reducers.js#L64
- https://github.com/indico/indico/blob/676dc3d1fede9bd4417982e79310753691be2472/indico/modules/events/editing/client/js/editing/timeline/util.js#L55

**Comment by ljharb** [comment]
cc @vedadeepta
